src/rt/array/construction.d

```diff
@@ -0,0 +1,165 @@
+/**
+ This module contains compiler support for constructing dynamic arrays
+
+  Copyright: Copyright Digital Mars 2000 - 2019.
+  License: Distributed under the
+       $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+     (See accompanying file LICENSE)
+  Source: $(DRUNTIMESRC rt/_array/_construction.d)
+*/
+module rt.array.construction;
+
+/**
+ * Does array initialization (not assignment) from another array of the same element type.
+ * Params:
+ *  to = what array to initialize
+ *  from = what data the array should be initialized with
+ * Returns:
+ *  The constructed `to`
+ * Bugs:
+ *  The functions `@safe` safety is faked.
+ */
+T[] _d_arrayctor(T)(return scope T[] to, scope const T[] from) @trusted
+{
+    import core.internal.traits : Unqual;
+    import core.stdc.string : memcpy;
+
+    // Force `enforceRawArraysConformable` to be `pure`
+    void enforceRawArraysConformable(const char[] action, in size_t elementSize, const void[] a1, const void[] a2, in bool allowOverlap = false)
+    {
+        import rt.util.array : enforceRawArraysConformable;
+        mixin("alias Type = " ~ typeof(&enforceRawArraysConformable).stringof ~ " pure;");
+        (cast(Type)&enforceRawArraysConformable)(action, elementSize, a1, a2, allowOverlap);
+    }
```

Comment:
I understand you need to make `enforceRawArraysConformable` but why this technique?  Can you just type it out instead of using a string mixin?